crates/pixi_build_types/src/project_model.rs

```diff
@@ -925,4 +917,109 @@ mod tests {
                 .contains_key(&TargetSelectorV1::Unix)
         );
     }
+
+    #[test]
+    fn test_hash_collision_bug_dependency_fields() {
+        // Test that moving dependencies between different dependency types produces different hashes
+
+        let mut deps = OrderMap::new();
+        deps.insert("python".to_string(), PackageSpecV1::Binary(Box::default()));
+
+        // Same dependency in host_dependencies
+        let target1 = TargetV1 {
+            host_dependencies: Some(deps.clone()),
+            build_dependencies: None,
+            run_dependencies: None,
+        };
+
+        // Same dependency in run_dependencies
+        let target2 = TargetV1 {
+            host_dependencies: None,
+            build_dependencies: None,
+            run_dependencies: Some(deps.clone()),
+        };
+
+        // Same dependency in build_dependencies
+        let target3 = TargetV1 {
+            host_dependencies: None,
+            build_dependencies: Some(deps.clone()),
+            run_dependencies: None,
+        };
+
+        let hash1 = calculate_hash(&target1);
+        let hash2 = calculate_hash(&target2);
+        let hash3 = calculate_hash(&target3);
+
+        assert_ne!(
+            hash1, hash2,
+            "Same dependency in host vs run should produce different hashes"
+        );
+        assert_ne!(
+            hash1, hash3,
+            "Same dependency in host vs build should produce different hashes"
+        );
+        assert_ne!(
+            hash2, hash3,
+            "Same dependency in run vs build should produce different hashes"
+        );
+
+        // Test with TargetsV1 as well
+        let targets1 = TargetsV1 {
+            default_target: Some(target1),
+            targets: None,
+        };
+
+        let targets2 = TargetsV1 {
+            default_target: Some(target2),
+            targets: None,
+        };
+
+        let targets_hash1 = calculate_hash(&targets1);
+        let targets_hash2 = calculate_hash(&targets2);
+
+        assert_ne!(
+            targets_hash1, targets_hash2,
+            "TargetsV1 should produce different hashes for different dependency types"
+        );
+    }
+
+    #[test]
+    fn test_hash_collision_bug_project_model() {
+        // Test the same issue in ProjectModelV1
+        let project1 = ProjectModelV1 {
+            name: "test".to_string(),
+            description: Some("test description".to_string()),
```

Comment:
Nitpick, but, is it possible to use `..Default` not sure, I had to read like 8 times to see that the structures were not equivalent.